app/http.php

```diff
@@ -36,11 +44,53 @@
         // 'document_root' => __DIR__.'/../public',
         // 'enable_static_handler' => true,
         'http_compression' => true,
+        'dispatch_func' => 'dispatch',
         'http_compression_level' => 6,
         'package_max_length' => $payloadSize,
         'buffer_output_size' => $payloadSize,
     ]);
 
+function dispatch(Server $server, $fd, $type, $data = null)
+{
+    global $workerNumber, $domains;
+
+    $safeThreadsPercent = intval(App::getEnv('_APP_SAFE_THREADS_PERCENT', 80)) / 100;
+
+    $safeThreads = (int) floor($workerNumber * $safeThreadsPercent);
+
+    $risky = false;
+    if ($data && str_contains($data, 'POST') && str_contains($data, '/executions')) {
```

Comment:
Wouldn't it be easier to first parse the payload and the set the conditions? It would also be more effective than the extensive usage in str_contains